service/controller/clusterapi/v27/resource/namespace/current.go

```diff
@@ -54,7 +55,7 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, nil
 	}
 
-	if namespace == nil && legacykey.IsDeleted(customObject) {
+	if namespace == nil && key.IsDeleted(&customObject) {
```

Comment:
I don't like it too much because the pointer is here required where it is not in all other places. At least the compiler support helps with that.